modular_skyrat/modules/microfusion/code/microfusion_cell_attatchments.dm

```diff
@@ -0,0 +1,131 @@
+/*
+MICROFUSION CELL UPGRADE ATTACHMENTS
+
+For adding unique abilities to microfusion cells. These cannot directly interact with the gun.
+*/
+
+/obj/item/microfusion_cell_attachment
+	name = "microfusion cell attachment"
+	desc = "broken"
+	icon = 'modular_skyrat/modules/microfusion/icons/microfusion_cells.dmi'
+	w_class = WEIGHT_CLASS_NORMAL
+	/// The overlay that will be automatically added, must be in the cells icon.
+	var/attachment_overlay_icon_state
+	/// Does this attatchment process with the cell?
+	var/processing_attachment = FALSE
+	/// How much stability does this cost the cell?
+	var/stability_impact = 0
+
+/obj/item/microfusion_cell_attachment/examine(mob/user)
+	. = ..()
+	. += span_notice("It has a stability impact factor of [stability_impact]%.")
+
+/obj/item/microfusion_cell_attachment/proc/run_attachment(obj/item/stock_parts/cell/microfusion/microfusion_cell)
+	SHOULD_CALL_PARENT(TRUE)
+	START_PROCESSING(SSobj, microfusion_cell)
+	microfusion_cell.instability += stability_impact
+	return
+
+/obj/item/microfusion_cell_attachment/proc/process_attachment(obj/item/stock_parts/cell/microfusion/microfusion_cell, delta_time)
+	return PROCESS_KILL
+
+/obj/item/microfusion_cell_attachment/proc/remove_attachment(obj/item/stock_parts/cell/microfusion/microfusion_cell)
+	SHOULD_CALL_PARENT(TRUE)
+	STOP_PROCESSING(SSobj, microfusion_cell)
+	microfusion_cell.instability -= stability_impact
+	return
+
+/*
+RECHARGABLE ATTACHMENT
+
+Allows the cell to be recharged at a gun recharger OR cell recharger.
+*/
+/obj/item/microfusion_cell_attachment/rechargeable
+	name = "rechargeable microfusion cell attachment"
+	desc = "Enables recharging on the microfusion cell its attatched to."
+	icon_state = "attachment_rechargeable"
+	attachment_overlay_icon_state = "microfusion_rechargeable"
+	/// The bonus charge rate by adding this attachment.
+	var/charge_rate = 100
+	var/initial_charge_rate = 0
```

Comment:
this might get confusing with `initial(charge_rate)`

Reply:
Not even needed!